### PR TITLE
MULE-13150: javax.xml.bind.DatatypeConverterImpl generates a class lo…

### PR DESCRIPTION
**Pull request by gsfernandes**
…ader leak when used by plugins

### CODE REVIEW

File: modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceInitializer.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+package org.mule.module.artifact.classloader;
+
+import org.mule.runtime.module.artifact.classloader.ResourceInitializer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Default implementation used for every Mule container.
+ */
```
Comment (pablokraan) [inline modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceInitializer.java]: Add `@since`

File: modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceInitializer.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  private final transient Logger logger = LoggerFactory.getLogger(getClass());
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void initialize() {
+    try {
+      Class.forName("javax.xml.bind.DatatypeConverterImpl");
```
Comment (pablokraan) [inline modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceInitializer.java]: Add a comment explaining why is this required

File: modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceInitializer.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import org.mule.runtime.module.artifact.classloader.ResourceInitializer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Default implementation used for every Mule container.
+ */
+public class DefaultResourceInitializer implements ResourceInitializer {
```
Comment (pablokraan) [inline modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceInitializer.java]: Be sure that this class is not exported

File: modules/artifact/src/main/java/org/mule/runtime/module/artifact/classloader/ResourceInitializer.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+package org.mule.runtime.module.artifact.classloader;
+
+/**
+ * Implementations of this class should take care of resources initialization with System Class Loader
+ * as if they are loaded with application's Class Loader it may leak memory after application is undeployment. Mule
+ * ensures to create an instance of this class with the System Class Loader.
+ */
+public interface ResourceInitializer {
```
Comment (pablokraan) [inline modules/artifact/src/main/java/org/mule/runtime/module/artifact/classloader/ResourceInitializer.java]: This interface is not needed (is implemented but never referenced otherwise)

File: modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceInitializer.java (hunk 3 above)
Comment (gsfernandes) [inline modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceInitializer.java]: Not exported as it is in the same package as DefaultResourceReleaser

File: modules/artifact/src/main/java/org/mule/runtime/module/artifact/classloader/ResourceInitializer.java (hunk 4 above)
Comment (gsfernandes) [inline modules/artifact/src/main/java/org/mule/runtime/module/artifact/classloader/ResourceInitializer.java]: Ok, interface not needed and removed.

File: modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceInitializer.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.module.artifact.classloader;
```
Comment (pablokraan) [inline modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceInitializer.java]: Package must have org.mule.runtime prefix
